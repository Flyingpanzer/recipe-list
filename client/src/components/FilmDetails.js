import React, { Fragment } from 'react';

const FilmDetails = ({ film }) => {
  return (
    <div className="col-md-8">
      <div>
        {film && (
          <Fragment>
            <h3>Film Details</h3>
            <table className="table table-bordered table-striped">
              <thead>
                <tr>
                  <th>Title</th>
                  <th>Release Year</th>
                  <th>Format</th>
                  <th>Stars</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>{film.filmTitle}</td>
                  <td>{film.filmYear}</td>
                  <td>{film.filmFormat}</td>
                  <td>{film.filmStars.join(', ')}</td>
                </tr>
              </tbody>
            </table>
          </Fragment>
        )}
      </div>
    </div>
  );
};

export default FilmDetails;
