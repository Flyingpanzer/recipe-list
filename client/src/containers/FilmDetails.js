import { connect } from "react-redux";

import FilmDetails from "../components/FilmDetails";

const mapStateToProps = ({ filmState }) => {
  return {
    film: filmState.film
  };
};

export default connect(mapStateToProps, null)(FilmDetails);
