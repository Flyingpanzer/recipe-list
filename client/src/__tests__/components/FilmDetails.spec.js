import React from "react";
import { mount } from "enzyme";

import FilmDetails from "../../components/FilmDetails";

const filmDetails = {
  filmTitle: "Stalker",
  filmYear: "1979	",
  filmFormat: "VHS",
  filmStars: "Alexander Kaidanovsky"
};
describe("SearchDetails component", () => {
  test("receive props and renders", () => {
    const wrapper = mount(<FilmDetails filmDetails={filmDetails} />);

    expect(wrapper).toMatchSnapshot();
  });
});
