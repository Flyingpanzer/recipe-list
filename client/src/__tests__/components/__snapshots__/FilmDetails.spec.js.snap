// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDetails component receive props and renders 1`] = `
ReactWrapper {
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
    "wrapInvoke": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "filmDetails": Object {
        "filmFormat": "VHS",
        "filmStars": "Alexander Kaidanovsky",
        "filmTitle": "Stalker",
        "filmYear": "1979	",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="col-md-8"
>
        <div />
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div />,
        "className": "col-md-8",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "filmDetails": Object {
          "filmFormat": "VHS",
          "filmStars": "Alexander Kaidanovsky",
          "filmTitle": "Stalker",
          "filmYear": "1979	",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="col-md-8"
>
          <div />
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "col-md-8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__linkedRoots__): Array [],
  Symbol(enzyme.__unrendered__): <FilmDetails
    filmDetails={
        Object {
            "filmFormat": "VHS",
            "filmStars": "Alexander Kaidanovsky",
            "filmTitle": "Stalker",
            "filmYear": "1979	",
          }
    }
/>,
  Symbol(enzyme.__updatedBy__): null,
}
`;
